protostar-rust/tests/e2e/running.rs

```diff
@@ -26,6 +28,56 @@ fn running_tests() {
             [PASS] test_my_test::test_my_test::test_four
             Running 1 test(s) from src/lib.cairo
             [PASS] [..]::test_fib
-            Tests: 5 passed, 1 failed
+            Tests: 6 passed, 1 failed
+        "#});
+}
+
+#[test]
+fn running_tests_with_filter() {
+    let temp = assert_fs::TempDir::new().unwrap();
+    temp.copy_from("tests/data/example_package", &["**/*"])
+        .unwrap();
+
+    let snapbox = runner();
+    let corelib = corelib_path();
+
+    snapbox
+        .current_dir(&temp)
+        .arg("two")
+        .args(["--corelib-path", corelib])
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"Collected 2 test(s) and 4 test file(s)
+            Running 0 test(s) from tests/without_prefix.cairo
+            Running 2 test(s) from tests/test_2.cairo
+            [PASS] test_2::test_2::test_two
+            [FAIL] test_2::test_2::test_two_failing 2 == 3
+            Running 0 test(s) from tests/test_my_test.cairo
+            Running 0 test(s) from src/lib.cairo
+            Tests: 1 passed, 1 failed
+        "#});
+}
+
+#[test]
+fn running_tests_with_non_matching_filter() {
+    let temp = assert_fs::TempDir::new().unwrap();
+    temp.copy_from("tests/data/example_package", &["**/*"])
+        .unwrap();
+
+    let snapbox = runner();
+    let corelib = corelib_path();
+
+    snapbox
+        .current_dir(&temp)
+        .arg("qwerty")
+        .args(["--corelib-path", corelib])
+        .assert()
+        .success()
+        .stdout_matches(indoc! {r#"Collected 0 test(s) and 4 test file(s)
+            Running 0 test(s) from tests/without_prefix.cairo
+            Running 0 test(s) from tests/test_2.cairo
+            Running 0 test(s) from tests/test_my_test.cairo
+            Running 0 test(s) from src/lib.cairo
+            Tests: 0 passed, 0 failed
```

Comment:
Okay, I did not think about that when implementing pretty printing xD We should probably "if" this print if there are 0 matching elements found + print that no tests were found if that's the case (I guess we should still return 0 exit code?)

Reply:
https://github.com/software-mansion/protostar/issues/2069

Reply:
Let's check how cargo test handles this. I think it's actually okay to have prints like this with non matching filter, it's consistent.